build-prybar.nix

```diff
@@ -1,17 +1,22 @@
 { language, buildInputs ? [] }:
 
-{ lib, buildGoModule, fetchFromGitHub, bash, pkg-config }:
+{ lib, buildGoModule, fetchFromGitHub, bash, pkg-config, runCommand, git }:
 
+let
+    gitSrc = builtins.filterSource
```

Comment:
Do we have to use `filterSource`?
Also minor nit, we might be able to do this, which seems slightly more idiomatic (and doesn't require `rec` anymore):
```nix
...

let

    src = ./.;

    revision = runCommand "get-rev" {
        nativeBuildInputs = [ git ];
        dummy = builtins.currentTime;
    } "GIT_DIR=${gitSrc}/.git git rev-parse --short HEAD | tr -d '\n' > $out";

in buildGoModule {
    pname = "prybar-${language}";
    version = builtins.readFile revision;

    inherit src;

    ...
}
```

Reply:
I couldn't find any way to do the equivalent (Copy files from another directory) with anything other than `filterSource` :thinking:. I'll try to explore your change and see if I can't simplify the code.

Reply:
Yeah, nix can't access file outside of the store. So trying to do something like 
```
revision = runCommand "get-rev" {
        nativeBuildInputs = [ git ];
        dummy = builtins.currentTime;
    } "GIT_DIR=${toString src}/.git git rev-parse --short HEAD | tr -d '\n' > $out";
```

Will trigger a not found error

Reply:
Ah okay I guess Nix uses what you did, but they offer a convenience function:
```nix
copyPathToStore = builtins.filterSource (p: t: true);
```
I think this should work
```nix
src = copyPathToStore ./.;
```

Reply:
I just remembered, the big reason we have to use `filterSource` over `fetchGit` is that `fetchGit` does not copy the .git folder.

Reply:
OH! I did not know this existed, let me switch to that

Reply:
Yeah we definitely don't want `fetchGit` anymore since that would pull the repo at the specified SHA instead of using the existing working directory.

Reply:
I'm being very nit-picky here (mainly because I'm still figuring this stuff out as well :smile: ), but it is possible to combine both `src` and `gitSrc` into the same thing? ie: `src = copyPathToStore ./.;`

Otherwise I _think_ we end up copying the whole working directory to the Nix store twice

Reply:
Let me try that real quick

Reply:
Seems like it works! :smile: I had assumed `src` used `./.` as a string rather than fetch the content, that's still somewhat confusing to me :laughing:

Reply:
haha yeah its a bit magic, I used `filterSource` on the `src` in the `pid1` derivation so we didn't have to copy some particularly large directories to the Nix store.

Reply:
Pushed all the changes, let me know if you have any other ideas for improvements :slightly_smiling_face: